plugins/e621.py

```diff
@@ -0,0 +1,113 @@
+# The MIT License (MIT)
+
+# Copyright (c) 2016 HeyItsShuga
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import logging
+import re
+import html
+from urllib.parse import urlsplit
+import traceback
+
+import json
+import requests
+import mimeparse
+import praw
+
+
+class e621Plugin:
+    """
+    Mirrors e621 images.
+    Created by /u/HeyItsShuga
+
+    """
+
+    def __init__(self, useragent: str, **options):
+        """Initialize the e621 importer.
+
+        :param useragent: The useragent to use for querying e621.
+        :param options: Other options in the configuration. Ignored.
+        """
+        self.log = logging.getLogger('lapis.e621')
+        self.headers = {'User-Agent': useragent}
+        self.regex = re.compile(r'^https?://(((?:www\.)?(?:static1\.)?((e621)|(e926))\.net/(data/.+/(?P<cdn_id>\w+))?(post/show/(?P<id>\d+)/?)?.*))$')
+
+    def import_submission(self, submission: praw.objects.Submission) -> dict:
+        """Import a submission from e621.
+
+        This function will define the following values in its return data:
+        - author: simply "an anonymous user on e621"
+        - source: The url of the submission
+        - importer_display/header
+        - import_urls
+
+        After we define that, we need to get the image. Since e621 has an API,
+        we use that to try to get the image if the image is a non-CDN URL. If it is
+        a CDN, we take the image directory and upload *that* to Imgur.
+
+        image_url is the variable of the image to upload.
+
+        :param submission: A reddit submission to parse.
+        """
+        print('e621 PLUGIN DEBUG - Initiated plugin.')
```

Comment:
`self.log` should be used - *not* `print`. That is what it's there for, after all.